src/Raven.Client/Documents/Session/Tokens/WhereToken.cs

```diff
@@ -372,11 +369,11 @@ public override void WriteTo(StringBuilder writer)
 
                     writer.Append(")");
                     break;
-                case WhereOperator.CmpXchgMatch:
-                    writer
-                        .Append(", $")
-                        .Append(ParameterName);
-                    writer.Append(")");
+                case WhereOperator.MethodCall:
+                    writer.Append(" = cmpxchg(\"");
```

Comment:
You are sending the parameters name as a string? 
We _want_ this as a parameter in queries, so users can send different things and it will generate the same query